### PR TITLE
Added MCSH5RecordingExtractor

### PR DESCRIPTION
**Pull request by alejoe91**
https://github.com/SpikeInterface/spikeinterface/issues/590

@jonasort could you test this? I don't have a test file so I might have missed something in the refactor. Actually it would be great if you could share a small test file with us :)

### CODE REVIEW

**Comment by jonasort** [comment]
@alejoe91 thank you! Did you receive the file I sent to you? I struggle to make it run on my own pc locally. Would you mind to give me some guidance on how to test it? Cheers!

**Comment by alejoe91** [comment]
@jonasort i didn't receive the file.. can you send it over again? How did you send it?

**Comment by jonasort** [comment]
@alejoe91 I sent it to your gmail address via a tresorit link (it is somewhat bigger than 1Gb). Maybe went to the spam folder?

**Comment by alejoe91** [comment]
It was indeed in spam! Will take a look at it next week. Happy Easter!

**Comment by jonasort** [comment]
Perfect! Happy Easter to you too!

**Comment by alejoe91** [comment]
@jonasort should be fixed now! Can you give it a go?

File: spikeinterface/extractors/mcsh5extractors.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        self._num_samples = int(num_frames)
+
+    def get_num_samples(self):
+        return self._num_samples
+    
+    def get_traces(self, 
+                   start_frame=None, 
+                   end_frame=None, 
+                   channel_indices=None):
+        stream = self._rf.require_group('/Data/Recording_0/AnalogStream/Stream_' + str(self._stream_id))
```
Comment (samuelgarcia) [inline spikeinterface/extractors/mcsh5extractors.py:76]: This should be done in init no ?
So avoid to fetch it at each chunk.

Comment (alejoe91) [reply spikeinterface/extractors/mcsh5extractors.py:76]: yep! Good catch! Fixing it now :)

**Comment by alejoe91** [comment]
@samuelgarcia can you merge this?

**Comment by samuelgarcia** [comment]
OK lets merge this but maybe it would be cool to make this hapenn in neo no ?
It is a trace format from a device almost all format are there.

**Comment by alejoe91** [comment]
@samuelgarcia I think we discussed it before and you said that the raw is already in Neo so there was no point to implement the h5 version there. But, you are welcome to port it there ;)

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**